File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def trav(self,root):
        if not root:
            return self.result
        self.trav(root.left)
        self.result.append(root.val)
        self.trav(root.right)
        return self.result
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.result=[]
        return self.trav(root)
        

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
void inorder(struct TreeNode* root,int* res,int* idx){
    if(root==NULL) return ;
    inorder(root->left,res,idx);
    res[(*idx)++]=root->val;
    inorder(root->right,res,idx);
 }
int* inorderTraversal(struct TreeNode* root, int* returnSize) {
    int* res=(int*)malloc(sizeof(int)*100);
    *returnSize=0;
    inorder(root,res,returnSize);
    return res;
}